pkg/maven/maven.go

```diff
@@ -107,6 +134,146 @@ func Evaluate(options *EvaluateOptions, expression string, command mavenExecRunn
+	}
+
+	var defines []string
+	if len(file) > 0 {
+		defines = append(defines, "-Dfile="+file)
+	} else {
+		defines = append(defines, "-Dfile="+pomFile)
+	}
+	defines = append(defines, "-DpomFile="+pomFile)
+	mavenOptionsInstall := ExecuteOptions{
+		Goals:   []string{"install:install-file"},
+		Defines: defines,
+		PomPath: pomFile,
+	}
+	_, err := Execute(&mavenOptionsInstall, command)
+	if err != nil {
+		return fmt.Errorf("failed to install maven artifacts: %w", err)
+	}
+	return nil
+}
+
+// InstallMavenArtifacts finds maven modules (identified by pom.xml files) and installs the artifacts into the local maven repository.
+func InstallMavenArtifacts(command mavenExecRunner, options EvaluateOptions) error {
+	return doInstallMavenArtifacts(command, options, newUtils())
+}
+
+func doInstallMavenArtifacts(command mavenExecRunner, options EvaluateOptions, utils mavenUtils) error {
+	err := flattenPom(command)
+	if err != nil {
+		return err
+	}
+
+	pomFiles, err := utils.glob(filepath.Join("**", "pom.xml"))
+	if err != nil {
+		return err
+	}
+
+	oldWorkingDirectory, err := utils.getwd()
+	if err != nil {
+		return err
+	}
+
+	for _, pomFile := range pomFiles {
+		log.Entry().Info("Installing maven artifacts from module: " + pomFile)
+		dir := path.Dir(pomFile)
+		err = utils.chdir(dir)
+		if err != nil {
+			return err
+		}
+
+		// Set pom path fix here because we cd'ed into the module's directory, so the pom is not in a subdirectory
+		options.PomPath = "pom.xml"
+		packaging, err := Evaluate(&options, "project.packaging", command)
+		if err != nil {
+			return err
+		}
+
+		if packaging == "pom" {
+			err = InstallFile("", "pom.xml", command)
+			if err != nil {
+				return err
+			}
+		} else {
+			err = installJarWarArtifacts(command, utils, options)
+			if err != nil {
+				return err
+			}
+		}
+
+		err = utils.chdir(oldWorkingDirectory)
+		if err != nil {
+			return err
+		}
+	}
+	return err
+}
+
+func installJarWarArtifacts(command mavenExecRunner, utils mavenUtils, options EvaluateOptions) error {
+	// Set pom path fix here because we cd'ed into the module's directory, so the pom is not in a subdirectory
+	options.PomPath = "pom.xml"
+	finalName, err := Evaluate(&options, "project.build.finalName", command)
+	if err != nil {
+		return err
+	}
+	if finalName == "" {
+		log.Entry().Warn("project.build.finalName is empty, skipping install of artifact. Installing only the pom file.")
+		err = InstallFile("", "pom.xml", command)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+	jarExists, err := utils.FileExists(jarFile(finalName))
+	if err != nil {
+		return err
+	}
+	warExists, err := utils.FileExists(warFile(finalName))
+	if err != nil {
+		return err
+	}
+
+	if jarExists {
+		err = InstallFile(jarFile(finalName), "pom.xml", command)
+		if err != nil {
+			return err
+		}
+	}
+
+	if warExists {
+		err = InstallFile(warFile(finalName), "pom.xml", command)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Can both jar and war files exist in parallel? I thought there is one target, as given by the packaging.

Reply:
Yes, you can have a jar and a war, and even multiple ones (think classes jar, sources jar, javadoc jar, ..), the question is which simplifications we can do here.

My idea was to go for the simple case where you have either a jar or a war following the naming convention. This might be too simple, but I don't see a suggestion in your comment.